plugin/src/software/aws/toolkits/eclipse/amazonq/util/QEclipseEditorUtils.java

```diff
@@ -139,7 +139,11 @@ private static String getOpenFilePath(final IEditorInput editorInput) {
         if (editorInput instanceof FileStoreEditorInput fileStoreEditorInput) {
             return fileStoreEditorInput.getURI().getPath();
         } else if (editorInput instanceof IFileEditorInput fileEditorInput) {
-            return fileEditorInput.getFile().getRawLocation().toOSString();
+            var file = fileEditorInput.getFile();
+            if (file.getRawLocation() == null && AbapUtil.isAbapFile(file)) {
```

Comment:
Consider flipping the order of the and operation or maybe we dont need to use the raw location at all and want to use  specific handling if the file is abapFile. Also add a comment indicating that special handling is needed for abap files because we are referencing the cached location it is stored with ADT plugins. 
```suggestion
            if (AbapUtil.isAbapFile(file)) {
                return AbapUtil.getSemanticCachePath(file.getFullPath().toOSString());
            }
```
> Maybe we dont need to use raw location

To clarify, we should check that this logic works and does not cause any regressions for when users have a regular ABAP file in their workspace, not using the ADT plugin but a general one. A general one should not try to resolve a semantic path if its not one .

Reply:
> Also add a comment indicating that special handling is needed for abap files because we are referencing the cached location it is stored with ADT plugins.

Please add this.